profiler/options.go

```diff
@@ -255,20 +255,18 @@ func defaultConfig() (*config, error) {
 	if v := os.Getenv("DD_VERSION"); v != "" {
 		WithVersion(v)(&c)
 	}
+
+	tags := make(map[string]string)
 	if v := os.Getenv("DD_TAGS"); v != "" {
-		sep := " "
-		if strings.Index(v, ",") > -1 {
-			// falling back to comma as separator
-			sep = ","
-		}
-		for _, tag := range strings.Split(v, sep) {
-			tag = strings.TrimSpace(tag)
-			if tag == "" {
-				continue
-			}
-			WithTags(tag)(&c)
-		}
+		tags = internal.ParseTagString(v)
 	}
+	for key, val := range internal.GetGitMetadataTags() {
+		tags[key] = val
+	}
+	for key, val := range tags {
+		WithTags(key + ":" + val)(&c)
```

Comment:
Small thing: technically, tags don't need to be "key:value", they can just be "key" ([ref](https://docs.datadoghq.com/getting_started/tagging/#define-tags)). So this is a bit of a change from the previous behavior. We don't have monitors based on profiling but possibly this could affect somebody's workflow if they use key-only tags in a profile query. Perhaps change this line to only do `key + ":" + val` if `val != ""`, and just do `key` otherwise?